cypress/integration/pages/articles/tests.js

```diff
@@ -101,38 +129,34 @@ export const testsThatFollowSmokeTestConfig = ({
         }
 
         it('should have an image copyright label with styling', () => {
-          cy.request(`${Cypress.env('currentPath')}.json`).then(({ body }) => {
-            const copyrightData = getBlockData('image', body);
-            const rawImageblock = getBlockByType(
-              copyrightData.model.blocks,
-              'rawImage',
-            );
-            const { copyrightHolder } = rawImageblock.model;
+          const copyrightData = getBlockData('image', articlesData);
+          const rawImageblock = getBlockByType(
+            copyrightData.model.blocks,
+            'rawImage',
+          );
+          const { copyrightHolder } = rawImageblock.model;
 
-            cy.get('figure')
-              .eq(0)
-              .within(() => {
-                if (copyrightHolder === 'BBC') {
-                  // If an image has a BBC copyright, the copyright holder (<p>) does not appear on images.
-                  // This is why we're asserting the value. If the copyright does not appear and is not
-                  // 'BBC' then it is clear there is an error with this component.
-                  cy.get('p[role="text"]').should('not.exist');
-                } else {
-                  cy.get('p[role="text"]')
-                    .should('be.visible')
-                    .and('contain', copyrightHolder);
-                }
-              });
-          });
+          cy.get('figure')
+            .eq(0)
+            .within(() => {
+              if (copyrightHolder === 'BBC') {
+                // If an image has a BBC copyright, the copyright holder (<p>) does not appear on images.
+                // This is why we're asserting the value. If the copyright does not appear and is not
+                // 'BBC' then it is clear there is an error with this component.
+                cy.get('p[role="text"]').should('not.exist');
+              } else {
+                cy.get('p[role="text"]')
+                  .should('be.visible')
+                  .and('contain', copyrightHolder);
+              }
+            });
         });
       }
 
       it('should have an inline link', () => {
-        cy.request(`${Cypress.env('currentPath')}.json`).then(({ body }) => {
-          if (body.data.article.metadata.language === 'en-gb') {
-            cy.get('main a');
-          }
-        });
+        if (articlesData.data.article.metadata.language === 'en-gb') {
+          cy.get('main a');
```

Comment:
not sure I understand this. All the other `get`s chain a `should` or similar at the end? This one has nothing chained and so by my (negligible) understanding will do nothing also? 

I am interested in learning more about cypress, so my understanding here may be way off the mark

Reply:
I prefer shoulds. They allow you to say if something is visible or just exists, and also negative assertions, what attributes elements have etc, and are also more explicit. This was written by someone else a loooong time ago, but I just double checked the cypress docs and it says:

cy.get() will automatically retry until the element(s) exist in the DOM

 cy.get() will automatically retry until all chained assertions have passed


So if th test never finds the 'main a' it will timeout and fail. I could change it to a should exist though.

Reply:
ah, awesome. Thanks!

Reply:
I have changed it to a should anyway